src/EditorFeatures/CSharpTest/QuickInfo/SyntacticQuickInfoSourceTests.cs

```diff
@@ -265,6 +265,254 @@ public async Task ScopeBrackets_10()
+@$"
+#region Start1
+#region Start2
+#endregion
+#end$$region", "#region Start1");
+        }
+
+        [WorkItem(56507, "https://github.com/dotnet/roslyn/issues/56507")]
+        [Fact, Trait(Traits.Feature, Traits.Features.QuickInfo)]
+        public async Task RegionEndShowsRegionStart_Nesting_2()
+        {
+            await TestAsync(
+@$"
+#region Start1
+#region Start2
+#end$$region
+#endregion", "#region Start2");
+        }
+
+        [WorkItem(56507, "https://github.com/dotnet/roslyn/issues/56507")]
+        [Fact, Trait(Traits.Feature, Traits.Features.QuickInfo)]
+        public async Task RegionEndShowsRegionStart_Blocks_1()
+        {
+            await TestAsync(
+@$"
+#region Start1
+#end$$region
+#region Start2
+#endregion", "#region Start1");
+        }
+
+        [WorkItem(56507, "https://github.com/dotnet/roslyn/issues/56507")]
+        [Fact, Trait(Traits.Feature, Traits.Features.QuickInfo)]
+        public async Task RegionEndShowsRegionStart_Blocks_2()
+        {
+            await TestAsync(
+@$"
+#region Start1
+#endregion
+#region Start2
+#end$$region", "#region Start2");
+        }
+
+        [WorkItem(56507, "https://github.com/dotnet/roslyn/issues/56507")]
+        [Fact, Trait(Traits.Feature, Traits.Features.QuickInfo)]
+        public async Task EndIfShowsIfCondition_1()
+        {
+            await TestAsync(
+@$"
+#if DEBUG
+#end$$if", "#if DEBUG");
+        }
+
+        [WorkItem(56507, "https://github.com/dotnet/roslyn/issues/56507")]
+        [Fact, Trait(Traits.Feature, Traits.Features.QuickInfo)]
+        public async Task EndIfShowsIfCondition_2()
+        {
+            await TestAsync(
+@$"
+#if DEBUG
+#else
+#end$$if", "#if DEBUG");
+        }
+
+        [WorkItem(56507, "https://github.com/dotnet/roslyn/issues/56507")]
+        [Fact, Trait(Traits.Feature, Traits.Features.QuickInfo)]
+        public async Task EndIfShowsElIfCondition()
+        {
+            await TestAsync(
+@$"
+#if DEBUG
+#elif RELEASE
+#end$$if", "#if DEBUG#elif RELEASE");
+        }
+
+        [WorkItem(56507, "https://github.com/dotnet/roslyn/issues/56507")]
+        [Fact, Trait(Traits.Feature, Traits.Features.QuickInfo)]
+        public async Task ElseShowsIfCondition()
+        {
+            await TestAsync(
+@$"
+#if DEBUG
+#el$$se
+#endif", "#if DEBUG");
+        }
+
+        [WorkItem(56507, "https://github.com/dotnet/roslyn/issues/56507")]
+        [Fact, Trait(Traits.Feature, Traits.Features.QuickInfo)]
+        public async Task ElseShowsElIfCondition_1()
+        {
+            await TestAsync(
+@$"
+#if DEBUG
+#elif RELEASE
+#el$$se
+#endif", "#if DEBUG#elif RELEASE");
+        }
+
+        [WorkItem(56507, "https://github.com/dotnet/roslyn/issues/56507")]
+        [Fact, Trait(Traits.Feature, Traits.Features.QuickInfo)]
+        public async Task ElseShowsElIfCondition_2()
+        {
+            await TestAsync(
+@$"
+#if DEBUG
+#elif RELEASE
+#elif DEMO
+#el$$se
+#endif", "#if DEBUG#elif RELEASE#elif DEMO");
```

Comment:
These are three different "relatedSpans". I don't know how QuickInfo handles such cases.

Reply:
can you check and show a picture of what this looks like?  thanks! :)

Reply:
I can't, unfortunately. I'm not able to F5 because of the VS 2022 preview requirement. Does it help to chow you the QuickInfoItem definition for this case?

Reply:
NOte: 2022 can be sxs installed with 2019.  That said, i'll pull this down locally to try out.

Reply:
I know. But I'm not allowed to do so. Thanks for taking the time to try this out!